### PR TITLE
AC-337:Add show password in the login window

### PR DESCRIPTION
**Pull request by DefCon-007**

### CODE REVIEW

**Comment by AdamGrzybkowski** [comment]
We should rather do it in material style. So instead of checkbox ant text use proper icon like https://material.io/icons/#ic_visibility and do it in one line.

**Comment by DefCon-007** [comment]
@AdamGrzybkowski  : Added the material style.

screenshot_20170130-191559
screenshot_20170130-191733

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java [hunk 1]
```diff
@@ -153,13 +159,34 @@ public void onClick(View v) {
...
+            }
+        });
     }
 
     private void initViewFields(View root) {
         mUrl = (EditText) root.findViewById(R.id.loginUrlField);
         mUsername = (EditText) root.findViewById(R.id.loginUsernameField);
         mUsername.setText(OpenMRS.getInstance().getUsername());
         mPassword = (EditText) root.findViewById(R.id.loginPasswordField);
+        mPassword.setCompoundDrawablesWithIntrinsicBounds(0 , 0,R.drawable.ic_password_hidden , 0);
```
Comment (AdamGrzybkowski) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java]: I would rather do this in xml so you can see layout without running the app

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java [hunk 2]
```diff
@@ -153,13 +159,34 @@ public void onClick(View v) {
                 forgotPassword();
             }
         });
+
+        mPassword.setOnTouchListener(new View.OnTouchListener() {
```
Comment (AdamGrzybkowski) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java]: why not onClickListener ?

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java [hunk 3]
```diff
@@ -153,13 +159,34 @@ public void onClick(View v) {
...
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
+                if (event.getAction() == MotionEvent.ACTION_DOWN) {
+                    if (event.getRawX() >= (mPassword.getRight() - mPassword.getCompoundDrawables()[2].getBounds().width())) {
+                        if (!showPasswordFlag) {
+                            mPassword.setInputType(InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD);
+                            mPassword.setCompoundDrawablesWithIntrinsicBounds(0, 0, R.drawable.ic_password_visible, 0);
```
Comment (AdamGrzybkowski) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java]: I would recommend you to create selector in drawable folder similar to [this ](https://github.com/openmrs/openmrs-contrib-android-client/blob/master/openmrs-client/src/main/res/drawable/sync_button_selector.xml). This way you will only have to set this field as selected

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java (hunk 2 above)
Comment (DefCon-007) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java]: @AdamGrzybkowski In onClickListener you have to click twice for edittext view, thats why I used onTouchListener.

**Comment by DefCon-007** [comment]
@AdamGrzybkowski : Made the requested changes. Please review.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java [hunk 4]
```diff
@@ -77,6 +81,8 @@
 
     private LoginValidatorWatcher loginValidatorWatcher;
 
+    private boolean showPasswordFlag = false ;
```
Comment (AdamGrzybkowski) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java]: @DefCon-007 Could you fix that ?

**Comment by DefCon-007** [comment]
@AdamGrzybkowski Fixed.

File: openmrs-client/src/main/res/drawable/showpassword_button_selector.xml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:state_checked="false"  android:drawable="@drawable/ic_password_hidden" />
```
Comment (AdamGrzybkowski) [inline openmrs-client/src/main/res/drawable/showpassword_button_selector.xml]: I think you have to revert the behavior. When the password is hidden the icon should not be crossed indicating that when you click it the password will be shown

Comment (DefCon-007) [inline openmrs-client/src/main/res/drawable/showpassword_button_selector.xml]: @AdamGrzybkowski : Switched the icon.

**Pull request merged**